modules/sdk-coin-bera/src/bera.ts

```diff
@@ -1,66 +1,53 @@
+/**
+ * @prettier
+ */
+import { BaseCoin, BitGoBase, common } from '@bitgo/sdk-core';
+import { BaseCoin as StaticsBaseCoin, coins, ethGasConfigs } from '@bitgo/statics';
+import {
+  AbstractEthLikeNewCoins,
+  recoveryBlockchainExplorerQuery,
+  TransactionBuilder as EthLikeTransactionBuilder,
+} from '@bitgo/abstract-eth';
+import { TransactionBuilder } from './lib';
+export class Bera extends AbstractEthLikeNewCoins {
   protected constructor(bitgo: BitGoBase, staticsCoin?: Readonly<StaticsBaseCoin>) {
     super(bitgo, staticsCoin);
-
-    if (!staticsCoin) {
-      throw new Error('missing required constructor parameter staticsCoin');
-    }
-
-    this._staticsCoin = staticsCoin;
   }
 
   static createInstance(bitgo: BitGoBase, staticsCoin?: Readonly<StaticsBaseCoin>): BaseCoin {
     return new Bera(bitgo, staticsCoin);
   }
 
-  /** @inheritDoc **/
-  getBuilder(): TransactionBuilderFactory {
-    return new TransactionBuilderFactory(coins.get(this.getChain()));
-  }
-
-  /** @inheritDoc **/
-  getBaseFactor(): string | number {
-    return 1e18;
-  }
-
-  /** @inheritDoc **/
-  isValidAddress(address: string): boolean {
-    return utils.isValidAddress(address) || utils.isValidValidatorAddress(address);
+  protected getTransactionBuilder(): EthLikeTransactionBuilder {
+    return new TransactionBuilder(coins.get(this.getBaseChain()));
   }
 
-  /** @inheritDoc **/
-  protected getPublicNodeUrl(): string {
-    return Environments[this.bitgo.getEnv()].beraNodeUrl;
+  /**
+   * Make a query to Arbiscan for information such as balance, token balance, solidity calls
+   * @param {Object} query key-value pairs of parameters to append after /api
+   * @returns {Promise<Object>} response from Arbiscan
+   */
+  async recoveryBlockchainExplorerQuery(query: Record<string, string>): Promise<Record<string, unknown>> {
+    const apiToken = common.Environments[this.bitgo.getEnv()].arbiscanApiToken;
+    const explorerUrl = common.Environments[this.bitgo.getEnv()].arbiscanBaseUrl;
+    return await recoveryBlockchainExplorerQuery(query, explorerUrl as string, apiToken);
   }
 
-  /** @inheritDoc **/
-  getDenomination(): string {
-    return BaseUnit.BERA;
-  }
-
-  /** @inheritDoc **/
-  getGasAmountDetails(): GasAmountDetails {
-    return {
-      gasAmount: GAS_AMOUNT,
-      gasLimit: GAS_LIMIT,
-    };
-  }
-
-  /** @inheritDoc **/
-  getKeyPair(publicKey: string): CosmosKeyPair {
-    return new KeyPair({ pub: publicKey });
-  }
-
-  /** @inheritDoc **/
-  getAddressFromPublicKey(publicKey: string): string {
-    return new KeyPair({ pub: publicKey }).getAddress();
+  /**
+   * Check whether gas limit passed in by user are within our max and min bounds
+   * If they are not set, set them to the defaults
+   * @param {number} userGasLimit user defined gas limit
+   * @returns {number} the gas limit to use for this transaction
+   */
+  setGasLimit(userGasLimit?: number): number {
```

Comment:
I think no need for adding this, this method is there in the abstractEthLikeNewCoins. This was added in arbitrum as there was a diff in values of gaslimit for arbitrum

Reply:
okay, we still not sure of the bera gas limit so removing it for now